cloudinit/config/cc_reset_rmc.py

```diff
@@ -0,0 +1,167 @@
+# (c) Copyright IBM Corp. 2020 All Rights Reserved
+#
+# Author: Aman Kumar Sinha <amansi26@in.ibm.com>
+#
+# This file is part of cloud-init. See LICENSE file for license information.
+
+
+"""
+Reset RMC
+------------
+**Summary:** reset rsct node id
+
+Reset RMC module is IBM PowerVM Hypervisor specific
+
+Reliable Scalable Cluster Technology (RSCT) is a set of software components,
+that  together provide a comprehensive clustering environment (RAS features)
+for IBM PowerVM based virtual machines. RSCT includes the Resource monitoring
+and control (RMC) subsystem. RMC is a generalized framework used for managing,
+monitoring, and manipulating resources. RMC runs as a daemon process on
+individual machines and needs creation of unique node id and restarts
+during VM boot.
+More details refer
+https://www.ibm.com/support/knowledgecenter/en/SGVKBA_3.2/admin/bl503_ovrv.htm
+
+This module handles
+- creation of the unique RSCT node id to every instance/virtual machine
+  and ensure once set, it isn't changed subsequently by cloud-init.
+  In order to do so, it restarts RSCT service.
+
+Prerequisite of using this module is to install RSCT packages.
+
+**Internal name:** per instance
+
+**Supported distros:** rhel, sles and ubuntu
+
+"""
+
+import json
+import os
+import time
+
+from cloudinit.settings import PER_INSTANCE
+from cloudinit import util
+from cloudinit import subp
+
+frequency = PER_INSTANCE
+PIDOF = 'pidof'
+RMCCTRL = '/usr/sbin/rsct/bin/rmcctrl'
+RECFGCT = '/usr/sbin/rsct/install/bin/recfgct'
+SRCMSTR = '/sbin/srcmstr'
+
+NODE_ID_FILE = '/etc/ct_node_id'
+
+SRCMSTR_TIMEOUT_SECONDS = 30
+
+
+def handle(name, _cfg, _cloud, log, _args):
+    global RMCCTRL, RECFGCT
+    with open('/run/cloud-init/instance-data.json') as data_file:
+        ds = json.load(data_file)
+        ds_value = ds['v1'].get('platform')
+
+    # Ensuring node id has to be generated only once during first boot
+    if ds_value == 'none':
+        log.debug('Skipping creation of new ct_node_id node')
+        return
+    required_tools = [RMCCTRL, RECFGCT]
+    for tool in required_tools:
+        if not os.path.isfile(tool):
+            log.debug('{} is not found, therefore trying {}.'
+                      .format(tool, tool.replace('/usr/sbin', '/opt')))
+            tool = tool.replace('/usr/sbin', '/opt')
+            if tool.split('/')[-1] == 'rmcctrl':
+                RMCCTRL = tool
+            if tool.split('/')[-1] == 'recfgct':
+                RECFGCT = tool
+            if not os.path.isfile(tool):
+                log.debug('%s is not found but is required, therefore not '
+                          'attempting to reset RMC.' % tool)
+                return
```

Comment:
Perhaps we could avoid all this path check if we just use system configuration like @smoser said above.